fv3core/fv3core/testing/translate_dyncore.py

```diff
@@ -145,4 +146,23 @@ def compute_parallel(self, inputs, communicator):
             inputs["pfull"],
             inputs["phis"],
         )
-        return super().compute_parallel(inputs, communicator)
+        self._base.make_storage_data_input_vars(inputs)
+        state = DycoreState.init_zeros(quantity_factory=self.grid.quantity_factory)
+        state.cappa = self.grid.quantity_factory.empty(
+            dims=[fv3util.X_DIM, fv3util.Y_DIM, fv3util.Z_DIM],
+            units="unknown",
+        )
+        for name, value in inputs.items():
+            if hasattr(state, name) and isinstance(state[name], fv3util.Quantity):
+                selection = tuple(slice(0, end) for end in value.shape)
+                state[name].storage[selection] = value
```

Comment:
I think `state[name].storage[:]` will also index everything regardless of shape

Reply:
Ah, this is actually needed because `value.shape` is not `state[name].storage.shape`. The storages have buffer points at the end of the axes in most cases. I'll add a comment to that effect.

Reply:
Updated.